src/Curiosity/Runtime/Order.hs

```diff
@@ -0,0 +1,28 @@
+module Curiosity.Runtime.Order
+  ( createOrder
+  , createOrderFull
+  , modifyOrders
+  ) where
+
+import qualified Control.Concurrent.STM        as STM
+import qualified Curiosity.Core                as Core
+import qualified Curiosity.Data                as Data
+import qualified Curiosity.Data.Order          as Order
+
+createOrder :: Core.StmDb -> STM (Either Order.Err Order.OrderId)
```

Comment:
All the functions here are in STM. This means this module should be in `Core`, not in `Runtime`. I'll still merge the PR as-is though to keep things going.